lib/logstash/outputs/tcp.rb

```diff
@@ -204,7 +176,38 @@ def server?
   end # def server?
 
   public
-  def receive(event)
-    @codec.encode(event)
-  end # def receive
+  def multi_receive_encoded(encoded)
+    if server?
+      @server_mutex.synchronize do
+        @client_threads.each do |client_thread|
+           encoded.each do |event,data|
+             client_thread[:client].write(data)
```

Comment:
It occurs to me that if you start up two TCP servers they would step on each other's use of `client_thread[:client]` no? Shouldn't the variable name include `self.id` to ensure uniqueness?